src/ucc/api/ucc.h

```diff
@@ -1097,22 +1131,113 @@ typedef struct ucc_ep_map_t {
  *  the fields is not set, the fields are not defined.
  *
  *
+ *
  *  @endparblock
  *
  */
 typedef struct ucc_team_params {
+
     uint64_t                mask;
+    /** @ref ucc_team_params.ordering is set to one the values defined by @ref
+      *  ucc_post_ordering_t
+      */
     ucc_post_ordering_t     ordering;
+
+    /** @ref ucc_team_params.outstanding_colls represents the number of outstanding non-blocking
+      * calls the user expects to post to the team. If the user posts more non-blocking
+      * calls than set, the behavior is undefined. If not set, there is no limit on
+      * the number of outstanding calls to be posted.
+      */
     uint64_t                outstanding_colls;
+
+    /** @ref ucc_team_params.ep The endpoint is a non-negative unique integer identifying the
+      * participant in the collective. If ep is not set, and @ref ucc_team_params.oob is not set, the
+      * library generates the ep. The generated ep can be queried using the @ref
+      * ucc_team_get_attr interface.
+      */
     uint64_t                ep;
+
+    /** @ref ucc_team_params.ep_list The endpoint list provides the list of eps participating to
+     *  create the team.
+     */
     uint64_t                *ep_list;
+
+    /** @ref ucc_team_params.ep_range can be either contiguous or not
+     *  contiguous. It is a hint to the library.
+     */
     ucc_ep_range_type_t     ep_range;
+
+    /** @ref ucc_team_params.team_size The team size is the number of participants in the team. If
+      * @ref ucc_team_params.oob is provided, the team size and @ref
+      * ucc_oob_coll.n_oob_eps should be the same.
+      */
     uint64_t                team_size;
+
+    /**
+      * @ref ucc_team_params.sync_type The options for sync_type are provided by @ref
+      * ucc_coll_sync_type_t
+      */
     ucc_coll_sync_type_t    sync_type;
+
+   /** @ref ucc_team_params.oob The signature of the function is defined by @ref
+     * ucc_oob_coll_t
+     * . The oob is used for exchanging information between the team
+     * participants during team creation. The user is
+     * responsible for implementing the oob operation. The relation between @ref
+     * ucc_team_params.ep and @ref ucc_oob_coll.oob_ep is defined as below:
+     *
+     * - When both are not provided. The library is responsible for generating the ep,
+     * which can be then queried via the @ref ucc_team_get_attr interface. This
+     * requires, however,
+     * ucc_params_t ep_map to be set and context created by @ref ucc_oob_coll.
+     * The behavior is
+     * undefined, when neither @ref ucc_team_params.ep or @ref
```

Comment:
When nothing is set, should we call it "undef" behavior or just call it error? We can add the check (in the team_create control path) and through the error.

Reply:
If it is error, we need to check and throw an error. If undefined, the onus is on the user.